/*
 * Copyright Â© 2023- Gimlet Labs, Inc.
 * All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Gimlet Labs, Inc. and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to Gimlet Labs, Inc. and its suppliers and
 * may be covered by U.S. and Foreign Patents, patents in process,
 * and are protected by trade secret or copyright law. Dissemination
 * of this information or reproduction of this material is strictly
 * forbidden unless prior written permission is obtained from
 * Gimlet Labs, Inc.
 *
 * SPDX-License-Identifier: Proprietary
 */

syntax = "proto3";

package gml.gem.calculators.opencv_cam.optionspb;

option go_package = "gimletlabs.ai/gimlet/src/gem/calculators/plugin/opencv_cam/optionspb;optionspb";

// This seems unused but the mediapipe subgraph options registration relies
// on mediapipe_options.proto being linked in.
// buf:lint:ignore IMPORT_USED
import "mediapipe/framework/mediapipe_options.proto";

message OpenCVCamSourceSubgraphOptions {
  string device_filename = 1;
}

message OpenCVCamSourceCalculatorOptions {
  string device_filename = 1;

  // Primarily for testing. 0 means no max.
  uint64 max_num_frames = 1000;
}
